api/clients/v2/accountant.go

```diff
@@ -5,115 +5,106 @@ import (
 	"fmt"
 	"math/big"
 	"sync"
+	"time"
 
 	disperser_rpc "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
 	"github.com/Layr-Labs/eigenda/core"
 	"github.com/Layr-Labs/eigenda/core/meterer"
 	gethcommon "github.com/ethereum/go-ethereum/common"
 )
 
-var requiredQuorums = []core.QuorumID{0, 1}
-
 type Accountant struct {
 	// on-chain states
-	accountID         gethcommon.Address
-	reservation       *core.ReservedPayment
-	onDemand          *core.OnDemandPayment
-	reservationWindow uint64
-	pricePerSymbol    uint64
-	minNumSymbols     uint64
+	accountID    gethcommon.Address
+	reservations map[core.QuorumID]*core.ReservedPayment
+	onDemand     *core.OnDemandPayment
+
+	paymentVaultParams *meterer.PaymentVaultParams
 
 	// local accounting
-	// contains an array of period records, with length of max(MinNumBins, numBins)
-	// numBins can be arbitrarily bigger than MinNumBins if the client wants to track more history in the cache
-	periodRecords     []PeriodRecord
+	// contains 3 bins; circular wrapping of indices
+	periodRecords     meterer.QuorumPeriodRecords
 	cumulativePayment *big.Int
 
 	// locks for concurrent access to period records and on-demand payment
 	periodRecordsLock sync.Mutex
 	onDemandLock      sync.Mutex
```

Comment:
is `onDemandLock` protecting `onDemand` or `periodRecords`? Or both?

Reply:
`onDemandLock` is for protecting `onDemand`. `periodRecordsLock` is for `periodRecords`. 
Per quorum is only for periodRecords (reservations). It will lock all the quorums' period records instead of one lock per `periodRecords`. updated comment in [commit](https://github.com/Layr-Labs/eigenda/pull/1664/commits/783ec1c7184a942bdfde1ef6ad4ffc611d96c776)